kernel/kernel-api/src/main/java/io/delta/kernel/TransactionBuilder.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel;
+
+import java.util.List;
+
+import io.delta.kernel.annotation.Evolving;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.types.StructType;
+
+/**
+ * Builder for creating a {@link Transaction} to mutate a Delta table.
+ *
+ * @since 3.2.0
+ */
+@Evolving
+public interface TransactionBuilder {
+    /**
+     * Set the new schema of the table. This is used when creating the table for the first time.
```

Comment:
nit: maybe "Set the schema of the table when creating a new table"? -- "new schema" kind of implies there was a former schema

Reply:
Same below in the param list